### PR TITLE
Build up stargazer timeseries beyond 40k gazers using periodic snapshots

### PR DESCRIPTION
**Pull request by jgehrcke**
WIP for https://github.com/jgehrcke/github-repo-stats/issues/76.

### CODE REVIEW

**Comment by Torantulino** [comment]
Thank you for working on this, excited to see it merged!

File: tests/analyze.bats [hunk 1]
```diff
@@ -112,3 +112,18 @@ setup() {
...
+@test "analyze.py: snapshots: some, vcagg: no, stars: some, forks: some, starsnaps: yes" {
+  run python analyze.py owner/repo tests/data/A/snapshots \
+    --resources-directory=resources \
+    --output-directory $BATS_TEST_TMPDIR/outdir \
+    --outfile-prefix "" \
+    --stargazer-ts-resampled-outpath stargazers-rs.csv \
+    --views-clones-aggregate-inpath tests/data/A/views_clones_aggregate.csv \
+    --fork-ts-inpath=tests/data/A/forks.csv \
+    --stargazer-ts-inpath=tests/data/A/stars.csv \
+    --stargazer-ts-snapshot-inpath=tests/data/A/stars-snapshots.csv
```
Comment (jgehrcke) [inline tests/analyze.bats:125]: That's the decisive addition in this test.

**Comment by jgehrcke** [comment]
> Thank you for working on this, excited to see it merged!

Thank you for the encouragement @Torantulino; it sometimes helps in the open-source world.

My plan is to merge something still today to get some further testing done (I have pipelines set up that run this action from its main branch. I hope not too many others have that :)).

Will then do a little more maintenance work, potentially do a little more thinking. Then I will invite you to test this out from the main branch, and then if things look good I will make a new release.

Screenshot from local testing (synthetic data):

Screenshot from 2023-09-30 22-44-10

**Comment by jgehrcke** [comment]
Tested this in action with
```
...
    steps:
      - name: GHRS
        uses: jgehrcke/github-repo-stats@HEAD
```
(note the `@HEAD`).

I think this has worked as expected.
- New file in data branch: https://github.com/jgehrcke/ghrs-test/blob/github-repo-stats/jgehrcke/github-repo-stats/ghrs-data/stargazer-snapshots.csv
- Plot looks good: https://jgehrcke.github.io/ghrs-test/jgehrcke/github-repo-stats/latest-report/report.html#stargazers

@ntindle @Torantulino @Swiftyos if you want to help testing this before release then please switch to `jgehrcke/github-repo-stats@HEAD` -- I am then happy to read logs / inspect outcome myself. Otherwise I understand if you want to wait for this being on the `RELEASE` branch. Cheers!

**Comment by ntindle** [comment]
I'll let the team know!

**Comment by Swiftyos** [comment]
Thank you, I've deployed it to our repo now.

**Pull request merged**